helpers/compile/plugins/tscPlugin.ts

```diff
@@ -83,8 +84,11 @@ export const tscPlugin: (emitTypes?: boolean) => esbuild.Plugin = (emitTypes?: b
         const tsconfig = require(`${process.cwd()}/${options.tsconfig}`) // tsconfig
         const typeOutDir = tsconfig?.compilerOptions?.outDir ?? '.' // type out dir
         const entryPoint = options.entryPoints?.[0].replace(/\.ts$/, '')
-        const typeOutPath = `${typeOutDir}/${entryPoint.replace(/^src\//, '')}`
         const bundlePath = options.outfile!.replace(/\.js$/, '')
+        const typeOutPath = [
+          `${process.cwd()}/${typeOutDir}/${entryPoint}.d.ts`,
+          `${process.cwd()}/${typeOutDir}/${entryPoint.replace(/^src\//, '')}.d.ts`,
+        ].filter(fs.existsSync)[0]
```

Comment:
Will it work in a clean checkout of the repo if this file doesn't exist yet?

Reply:
I think it should as the previous step generated the necessary files. This piece of code specifically helps to find the output whether it is generated and contains the `src` root folder in `dist` or not. This is inconsistent in TS 4, in theory only one path should work, but I think it was fixed in TS 5.

Reply:
Ah, I missed that part and was a bit confused, thank you!